apps/examples/fs_helper/procfs_test.c

```diff
@@ -0,0 +1,161 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/**
+ * @file procfs_test.c
+ * @brief This is test application for proc file system.
+ */
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <stdio.h>
+#include <fcntl.h>
+#include <errno.h>
+
+#include <tinyara/config.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+#define BUFFER_SIZE 1024
+/****************************************************************************
+ * Private data
+ ****************************************************************************/
+struct pthread_arg {
+	int argc;
+	char **argv;
+};
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+static int read_file(const char *srcpath)
+{
+	int nbytesread;
+	int rdfd;
+	static char buffer[BUFFER_SIZE];
+
+	/* Open the source file for reading */
+
+	rdfd = open(srcpath, O_RDONLY);
+	if (rdfd < 0) {
+		printf("ERROR: Failed to open %s for reading: %s\n", srcpath, strerror(errno));
+		return -1;
+	}
+
+	/* read the file */
+
+	for (;;) {
+		do {
+			nbytesread = read(rdfd, buffer, BUFFER_SIZE);
+			if (nbytesread == 0) {
+				/* End of file */
+
+				close(rdfd);
+				return OK;
+			} else if (nbytesread < 0) {
+				/* EINTR is not an error (but will still stop the copy) */
+
+				printf("ERROR: Read failure: %s\n", strerror(errno));
+				return -1;
+			}
+			printf("read file string[%s]", buffer);
+
+		} while (nbytesread <= 0);
+	}
+	close(rdfd);
+	return OK;
+}
+
+int procfs_commands(void *args)
+{
+	char filename[100] = "/proc/0/status";
+	int ret;
+//  ret = mount(NULL, "/proc", "procfs", 0, NULL);
+//  if(ret < 0){
+//    printf("procfs mount error[%d]\n",ret);
+//    return -1;
+// }
+
+	ret = read_file(filename);
+	if (ret < 0) {
+		printf("procfs error!\n");
+	} else {
+		printf("success procfs test!\n");
+	}
+	if (umount("/proc") < 0) {
+		printf("umount procfs fail\n");
+		return -1;
+	}
+	return 0;
+}
+
+/****************************************************************************
+ * fs_command_main
+ ****************************************************************************/
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+int procfs_test_main(int argc, char **argv)
```

Comment:
Why don't you register profs_command directly? In buitin functionality, we already support setting of specific PRI and STACK.